hostap: support enterprise and DPP

fengming-ye opened the pull request:
Change cmake and kconfig for enterprise and DPP support.

fengming-ye committed:
mbedtls: add kconfig options

Add MBEDTLS_NIST_KW_C, MBEDTLS_DHM_C and X509 CRL, CSR options.

Signed-off-by: Fengming Ye <frank.ye@nxp.com>

fengming-ye commented:
There might be a CI issue in Complicance check fail.
warning: undefined symbol MBEDTLS_MAC_CMAC_ENABLED:
but MBEDTLS_MAC_CMAC_ENABLED is defined in modules/mbedtls/Kconfig.tls-generic

krish2718 reviewed (commented):
From a quick look changes LGTM, but can you please divide in to smaller commits with appropriate commit logs? 

E.g.

- NO_PBKDF2
- Removing blobs and why
- Adding CRC32
- Moving existing crypto
- Adding DPP
- Adding ALT crypto
...

fengming-ye committed:
hostap: add crypto backend alt for enterprise and DPP

Add kconfig CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT to get more mbedtls
functionality for enterprise and DPP.
Split cmake sources related to hostap SME and crypto backend.
Default backend CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO use internal crypto
and some mbedtls apis.
Backend CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT use most mbedtls apis
and PSA apis, except some apis no longer supported in mbedtls 3.x,
or called in the middle of hostap and mbedtls.

Signed-off-by: Fengming Ye <frank.ye@nxp.com>

fengming-ye committed:
hostap: add PSA apis support for mbedtls 3.x

Add Platform Secure Architecture support support to use
HW acceleration, which needs to be called under PSA driver wrapper
in mbedtls 3.x.

Signed-off-by: Fengming Ye <frank.ye@nxp.com>

fengming-ye committed:
hostap: add DPP support

Add kconfig and cmake to support DPP (Easy Connect).

Signed-off-by: Fengming Ye <frank.ye@nxp.com>

fengming-ye committed:
hostap: remove CONFIG_NO_PBKDF2 and CONFIG_NO_CONFIG_BLOBS in cmake

Remove CONFIG_NO_PBKDF2 and CONFIG_NO_CONFIG_BLOBS definition
and let them be decided in kconfig.
CONFIG_NO_PBKDF2 is default y when crypto backend is
WIFI_NM_WPA_SUPPLICANT_CRYPTO_NONE.
CONFIG_NO_CONFIG_BLOBS is default y when both DPP and enterprise disable.

Signed-off-by: Fengming Ye <frank.ye@nxp.com>

fengming-ye committed:
hostap: add enterprise source files in cmake

Add enterprise needed source files.

Signed-off-by: Fengming Ye <frank.ye@nxp.com>

fengming-ye commented:
> From a quick look changes LGTM, but can you please divide in to smaller commits with appropriate commit logs?
> 
> E.g.
> 
> * NO_PBKDF2
> * Removing blobs and why
> * Adding CRC32
> * Moving existing crypto
> * Adding DPP
> * Adding ALT crypto
>   ...

updated pls check

fengming-ye commented:
There might be a CI issue in compliance check fail
 Undefined Kconfig symbols:

 warning: undefined symbol MBEDTLS_MAC_CMAC_ENABLED:

But MBEDTLS_MAC_CMAC_ENABLED is defined in modules/mbedtls/Kconfig.tls-generic like other symbols.

dleach02 commented:
> MBEDTLS_MAC_CMAC_ENABLED

@fengming-ye, this changed in a PR that merged yesterday (things move fast) [#73267]. All the MBED_MAC_<algorithm>_ENABLED have changed to just MBED_<algorithm>

dleach02 commented on modules/hostap/Kconfig:
```diff
@@ -125,11 +127,37 @@ config WIFI_NM_WPA_SUPPLICANT_CRYPTO
+	select MBEDTLS_CIPHER_MODE_CBC_ENABLED
+	select MBEDTLS_ECP_C
+	select MBEDTLS_ECP_ALL_ENABLED
+	select MBEDTLS_MAC_CMAC_ENABLED
```
Change this to MBEDTLS_CMAC

fengming-ye commented:
Hi @jukkar  @krish2718 
Here is kconfig and cmake changes for WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT
Pls also help review.

Hi @ceolin @d3zd3z @ithinuel 
Here is some kconfig adds for MBEDTLS_CONFIG macros.
Pls help review.

Thanks.

jukkar reviewed (approved):
LGTM
